decoder.go

```diff
@@ -22,12 +22,16 @@ func (s *publication) Retry() int64  { return s.retry }
 // LastEventID is from a separate interface, EventWithLastID
 func (s *publication) LastEventID() string { return s.lastEventID }
 
+// EnvironmentID is from a separate interface, EventWithEnvironmentID
+func (s *publication) EnvironmentID() string { return s.environmentID }
```

Comment:
I am a little hesitant to have something this LD specific in this layer. I believe our approach has been to more generally expose headers at this layer, and then make it environment ID specific up the stack.

Reply:
Such as:
https://github.com/launchdarkly/dotnet-eventsource/blob/07477d2abca12e70cd66a94059119dd7379d3699/src/LaunchDarkly.EventSource/EventSourceOpenedEventArgs.cs#L23

https://github.com/launchdarkly/rust-eventsource-client/blob/c8c6093626d2d1fcacfa7393c8296ba94c12540b/eventsource-client/src/client.rs#L488